src/controller/CHIPDevice.cpp

```diff
@@ -743,7 +743,8 @@ CHIP_ERROR Device::SendSubscribeAttributeRequest(app::AttributePathParams aPath,
 
     // The application context is used to identify different requests from client application the type of it is intptr_t, here we
     // use the seqNum.
-    app::ReadPrepareParams params(GetSecureSession());
+    VerifyOrReturnError(GetSecureSession().HasValue(), CHIP_ERROR_INCORRECT_STATE);
+    app::ReadPrepareParams params(GetSecureSession().Value());
```

Comment:
@yunhanw-google Are you ok with the changes to these lines?

Reply:
Are you shifting the logic from asserting if `mSecureSession` has no value (i.e no secure session) TO  just returning an error instead?

Reply:
Nvm - saw your description. Change makes sense, but I'm curious as to how you use the now-returned-error usefully in OTA?

Reply:
This PR is necessary because `mSecureSession` was changed from being a `SessionHandle` to being of type `chip::Optional<SessionHandle>`.

The change above is just to guard against the case where `SendSubscribeAttributeRequest()` is called but `mSecureSession` has not yet been set. Because `GetSecureSession()` would call `mSecureSession.Value()` which aborts is the value doesn't exist.

OTA code doesn't call `SendSubscribeAttributeRequest()` so I don't need to handle the error